src/main/java/io/papermc/lib/PaperLegacyEnvironment.java

```diff
@@ -37,4 +39,14 @@ public boolean isPaper() {
         }
         return future;
     }
+
+    @Override
+    public BlockState getBlockState(Block block, boolean useSnapshot) {
+        return block.getState(useSnapshot);
+    }
+
+    @Override
+    public boolean updateBlockState(BlockState blockState, boolean force, boolean applyPhysics) {
```

Comment:
I don't know if this is intuitive. What if you passed a block state that WAS a snapshot?

Maybe we should just optimize paper to ignore updates on non snapshots instead of a PaperLib API?

Reply:
Yeah that’d work too, I was trying to come up with a good way of not updating snapshots without a “isPaper” check in the plugin, but adding it to paper itself would be a good idea.

Reply:
Done, https://github.com/PaperMC/Paper/pull/1535/files